dot/parachain/candidate-validation/candidate_validation_test.go

```diff
@@ -508,3 +509,125 @@ func TestCandidateValidation_processMessageValidateFromChainState(t *testing.T)
+
+	mockInstanceShortTimeout := NewMockInstance(ctrl)
+	mockInstanceShortTimeout.EXPECT().ParachainHostValidationCodeByHash(common.MustHexToHash("0x0404")).Return(
+		&validationCode, nil)
+	mockInstanceShortTimeout.EXPECT().ParachainHostSessionIndexForChild().Return(parachaintypes.SessionIndex(3), nil)
+	executionParamsShortTimeout := parachaintypes.ExecutorParams{}
+	timeoutShort := parachaintypes.PvfPrepTimeout{
+		PvfPrepTimeoutKind: func() parachaintypes.PvfPrepTimeoutKind {
+			kind := parachaintypes.NewPvfPrepTimeoutKind()
+			if err := kind.SetValue(parachaintypes.Precheck{}); err != nil {
+				panic(err)
+			}
+			return kind
+		}(),
+		Millisec: 1,
+	}
+	timeoutShortParam := parachaintypes.NewExecutorParam()
+	err = timeoutShortParam.SetValue(timeoutShort)
+	require.NoError(t, err)
+	executionParamsShortTimeout = append(executionParamsShortTimeout, timeoutShortParam)
+	mockInstanceShortTimeout.EXPECT().ParachainHostSessionExecutorParams(parachaintypes.SessionIndex(3)).Return(
+		&executionParamsShortTimeout, nil)
+
+	mockBlockState := NewMockBlockState(ctrl)
+	mockBlockState.EXPECT().GetRuntime(common.MustHexToHash("0x01")).Return(nil, fmt.Errorf("runtime not found"))
+	mockBlockState.EXPECT().GetRuntime(common.MustHexToHash("0x02")).Return(mockInstance, nil)
+	mockBlockState.EXPECT().GetRuntime(common.MustHexToHash("0x03")).Return(mockInstanceExecutorError, nil)
+	mockBlockState.EXPECT().GetRuntime(common.MustHexToHash("0x04")).Return(mockInstanceShortTimeout, nil)
+
+	tests := map[string]struct {
+		msg            PreCheck
+		expectedResult PreCheckOutcome
+	}{
+		"validation_code_not_found": {
+			msg: PreCheck{
+				RelayParent: common.MustHexToHash("0x01"),
+			},
+			expectedResult: PreCheckOutcomeFailed,
+		},
+		"invalid_executor_params": {
+			msg: PreCheck{
+				RelayParent:        common.MustHexToHash("0x03"),
+				ValidationCodeHash: parachaintypes.ValidationCodeHash(common.MustHexToHash("0x04")),
+			},
+			expectedResult: PreCheckOutcomeInvalid,
+		},
+		"precheck_timeout": {
+			msg: PreCheck{
+				RelayParent:        common.MustHexToHash("0x04"),
+				ValidationCodeHash: parachaintypes.ValidationCodeHash(common.MustHexToHash("0x0404")),
+			},
+			expectedResult: PreCheckOutcomeFailed,
+		},
+		"happy_path": {
+			msg: PreCheck{
+				RelayParent:        common.MustHexToHash("0x02"),
+				ValidationCodeHash: candidate.Descriptor.ValidationCodeHash,
+			},
+			expectedResult: PreCheckOutcomeValid,
+		},
+	}
+	for name, tt := range tests {
+		tt := tt
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			toSubsystem := make(chan any)
+			candidateValidationSubsystem := CandidateValidation{
+				pvfHost:    newValidationHost(),
+				BlockState: mockBlockState,
+			}
+			defer candidateValidationSubsystem.Stop()
+
+			go candidateValidationSubsystem.Run(context.Background(), toSubsystem)
+
+			sender := make(chan PreCheckOutcome)
+			tt.msg.ResponseSender = sender
+
+			toSubsystem <- tt.msg
+			result := <-sender
+
```

Comment:
You are just testing one method. Which returns the value that you want to compare.
This method doesn't even use any channel if I am not wrong.
```suggestion
			candidateValidationSubsystem := CandidateValidation{
				pvfHost:    newValidationHost(),
				BlockState: mockBlockState,
			}

			result := candidateValidationSubsystem.precheckPvF(tt.msg.RelayParent, tt.msg.ValidationCodeHash)
```

Reply:
Good point, I've updated this.